<!-- Launches shelf detection-->
<launch>

 <arg name="debug" default="true" /> <!-- Whether to show debug visualizations. -->
 <param name="/use_sim_time" value="false"/> <!-- Use for testing with bag files. -->

  <!-- laser scan assembler -->
 <node pkg="laser_assembler" type="laser_scan_assembler" name="scan_assembler" respawn="true" output="screen">
     <remap from="scan" to="/tilt_scan" />
     <param name="max_clouds" value="250"/>
     <param name="fixed_frame" value="/base_footprint"/> <!-- pay attention to this reference system -->
 </node> 


  <node ns="perception" pkg="pr2_pick_perception" type="shelf_localization_service_node_lidar" name="shelf_obj_detector" respawn="true" output="screen">
   <param name="db" value="$(find pr2_pick_perception)/config/db_shelf.txt"/>

   <param name="iter" value="1000"/>
   <param name="minClusterSize" value="2350"/>
   <param name="radius_search" value="0.015"/>

   <param name="DistanceThreshold" value="0.07"/>
   <param name="PlanesegThres" value="0.03"/>   <!-- was 0.05 for non-hose, to remove wall behind valve -->
   <param name="PlaneSize" value="10000"/> <!-- WAS 4500,50000 -->
   <param name="depthplane" value="1.5" /> <!-- removes points further than this limit -->
   <param name="highplane" value="0.75" /> <!-- removes points higher than this limit -->
   <param name="ICP2D" value="true" /> 

  <!--  <param name="ModelReference" value="/head_mount_kinect_rgb_optical_frame" /> -->

   <param name="RobotReference" value="/base_footprint" />
   <param name="WorldReference" value="/base_footprint" />

   <param name="manual_segmentation" value="false" />
   <param name="pca_alignment" value="false"/>
   <param name="debug" value="$(arg debug)"/>

 
  </node>




</launch>
